servicetalk-concurrent-api/src/testFixtures/java/io/servicetalk/concurrent/api/TestExecutor.java

```diff
@@ -325,7 +334,29 @@ private RunnableWrapper(final Runnable delegate) {
 
         @Override
         public void run() {
-            delegate.run();
+            Thread current = Thread.currentThread();
+            String oldName = current.getName();
+            current.setName("TestExecutor");
```

Comment:
Consider adding a `factoryCount` for each new instance of `TestExecutor`, similar to how we generate names in `DefaultThreadFactory`. The `threadCount` probably is not necessary, but the `factoryCount` can help to distinguish between multiple instances of `TestExecutor`.